cpu/stm32/stmclk/stmclk_gx.c

```diff
@@ -191,73 +220,83 @@ void stmclk_init_sysclk(void)
+        RCC->CR |= RCC_CR_PLLON;
+        while (!(RCC->CR & RCC_CR_PLLRDY)) {}
+    }
+
+    /* Configure SYSCLK */
+    if (IS_ACTIVE(CONFIG_USE_CLOCK_HSE)) {
 #if defined(CPU_FAM_STM32G0)
-    RCC->CFGR = (RCC_CFGR_SW_HSE | CLOCK_AHB_DIV | CLOCK_APB1_DIV);
+        RCC->CFGR = (RCC_CFGR_SW_HSE | CLOCK_AHB_DIV | CLOCK_APB1_DIV);
 #elif defined(CPU_FAM_STM32G4)
-    RCC->CFGR = (RCC_CFGR_SW_HSE | CLOCK_AHB_DIV | CLOCK_APB1_DIV | CLOCK_APB2_DIV);
-#endif
-    while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_HSE) {}
-
-#elif CONFIG_USE_CLOCK_PLL
-#if CONFIG_BOARD_HAS_HSE
-    /* if configured, we need to enable the HSE clock now */
-    RCC->CR |= RCC_CR_HSEON;
-    while (!(RCC->CR & RCC_CR_HSERDY)) {}
+        RCC->CFGR = (RCC_CFGR_SW_HSE | CLOCK_AHB_DIV | CLOCK_APB1_DIV | CLOCK_APB2_DIV);
 #endif
-    /* now we can safely configure and start the PLL */
-    RCC->PLLCFGR = (PLL_SRC | PLL_M | PLL_N | PLL_R | RCC_PLLCFGR_PLLREN);
-    RCC->CR |= RCC_CR_PLLON;
-    while (!(RCC->CR & RCC_CR_PLLRDY)) {}
-
-#if CLOCK_AHB > MHZ(80)
-    /* Divide HCLK by before enabling the PLL */
-    RCC->CFGR |= RCC_CFGR_HPRE_DIV2;
+        RCC->CFGR |= RCC_CFGR_SW_HSE;
+        while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_HSE) {}
+    }
+    else if (IS_ACTIVE(CONFIG_USE_CLOCK_PLL)) {
+#if defined(CPU_FAM_STM32G4)
+        if (CLOCK_AHB > MHZ(80)) {
+            /* Divide HCLK by 2 before enabling the PLL */
+            RCC->CFGR |= RCC_CFGR_HPRE_DIV2;
+        }
 #endif
 
-    /* now that the PLL is running, we use it as system clock */
-    RCC->CFGR |= RCC_CFGR_SW_PLL;
-    while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_PLL) {}
-
-#if CLOCK_AHB > MHZ(80)
-    /* Wait 1us before switching back to full speed */
-    /* Use volatile to prevent the compiler from optimizing the loop */
-    volatile uint8_t count = CLOCK_CORECLOCK / MHZ(1);
-    while (count--) {}
-    RCC->CFGR &= ~RCC_CFGR_HPRE_DIV2;
-#endif
+        /* now that the PLL is running, we use it as system clock */
+        RCC->CFGR |= RCC_CFGR_SW_PLL;
+        while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_PLL) {}
 
+#if defined(CPU_FAM_STM32G4)
+        if (CLOCK_AHB > MHZ(80)) {
+            /* Wait 1us before switching back to full speed */
+            /* Use volatile to prevent the compiler from optimizing the loop */
+            volatile uint8_t count = CLOCK_CORECLOCK / MHZ(1);
+            while (count--) {}
+            RCC->CFGR &= ~RCC_CFGR_HPRE_DIV2;
+        }
 #endif
+    }
 
-    stmclk_disable_hsi();
-    irq_restore(is);
+    if (!IS_ACTIVE(CLOCK_ENABLE_HSI)) {
+        /* Disable HSI only if not used */
+        stmclk_disable_hsi();
+    }
 
-#ifdef MODULE_PERIPH_HWRNG
-    /* HWRNG is clocked by HSI48 so enable this clock when the peripheral is used */
-    RCC->CRRCR |= RCC_CRRCR_HSI48ON;
-    while (!(RCC->CRRCR & RCC_CRRCR_HSI48RDY)) {}
+#if defined(CPU_FAM_STM32G4)
+    if (IS_USED(MODULE_PERIPH_HWRNG)) {
```

Comment:
Wouldn't that also work

```suggestion
#if IS_USED(MODULE_PERIPH_HWRNG)
```

Reply:
HWRNG is only available on G4.

Reply:
So we don't need the CPU_FAM check at all :wink: 

I just remember porting samd5x was made pretty painful by hunting all the `CPU_FAM` checks that would silently fail and I was happy to get rid of most of them.

Reply:
> I just remember porting samd5x was made pretty painful by hunting all the CPU_FAM checks

I also know that story, I encountered it with l0, gx and more recently l5. Adding a new family always require these kind of changes, anyway. But they are done once.

In this particular block, I have to verify the RTT compatibility with g0 (#15090 was merged in the meantime with RTT support added and there was no RTT support for G0 when I opened this one)